core/system/src/mpu/vmpu_freescale_k64_mem.c

```diff
@@ -284,10 +351,15 @@ void vmpu_mem_init(void)
         HALT_ERROR(SANITY_CHECK_FAILED, "failed setting up box FLASH (%i)\n", res);
 
     /* initial primary box ACL's */
-    vmpu_mem_switch(0, 0);
-
-    /* mark all primary box ACL's as static */
-    g_mem_acl_user = g_mem_acl_count;
+    rgd = g_mem_box[0].acl;
+    /* Copy ACLs [0, 1] into MPU regions [_1_, _2_]. Region 0 is the background region! */
+    for(t = 1; t < K64F_MPU_REGIONS_RESERVED; t++, rgd++) {
+        word = MPU->WORD[t];
+        word[0] = rgd->word[0];
```

Comment:
@niklas-arm : instead of the array-assignment - in case there's not a struct for a single MPU slot provided by Freescale, please create one and use names instead of array indexes (with volatile members etc.)